docs: provide guidance on shared workspaces (#21214)
coder/coder: docs/user-guides/workspaces-shared.md

# Shared Workspaces

> Multiple users can securely connect to a single Coder workspace for debugging, programming, and code review. 

Reviewer: I'm not sure code review is really a use case people will want from shared workspaces

> (Insert screenshot of UI here)

Reviewer: ```suggestion
<!-- Insert screenshot of UI here -->
```

same here

## Features

Workspace sharing is available to all Coder users by default, but platform admins with a Premium subscription can choose to disable sharing within their organizations. 

Owners of a workspace can grant access to other users or groups with scoped roles. 

This is helpful in a number of scenarios, including: 

> - Developers can do ad-hoc debugging, pair programming, and code reviews.

Reviewer: ```suggestion
- Developers can do ad-hoc debugging or pair programming.
```

- A workspace can be owned by a group of users for QA, on-call rotations, or shared staging.
- AI workflows where an agent prepares a workspace and a developer takes over to review or finalize the work (ex. with [Coder Tasks](https://coder.com/docs/ai-coder/tasks).)

## Getting Started

Workspaces can be shared through either the Coder CLI or UI. 

> Before you begin, ensure that you have a version of Coder with workspace sharing enabled and that your account has permission to share workspaces. This is true by default if you are an OSS user, but Premium users are subject to organization-specific restrictions.

Reviewer: ```suggestion
Before you begin, ensure that you have a version of Coder with workspace sharing enabled and that your account has permission to share workspaces. This is true by default if you are an OSS user, but deployments with Premium licenses may be restricted by admins.
```
